f/mapClickTeleport/fn_mapClickTeleportAction.sqf

```diff
@@ -0,0 +1,76 @@
+// F3 - Mission Maker Teleport
+// Credits: Please see the F3 online manual (http://www.ferstaberinde.com/f3/en/)
+// ====================================================================================
+
+// SET KEY VARIABLES
+
+f_var_mapClickTeleport_telePos = nil;
+f_var_mapClickTeleport_MapClosed = false;
+if (isNil "f_var_mapClickTeleport_Used") then {f_var_mapClickTeleport_Used = 0};
+
+// ====================================================================================
+// Eventhandler to detect closing of the map
+// in case the player has not clicked on the map to select a teleport position.
+
+f_var_mapClickTeleport_EHMap = addMissionEventHandler ["Map", {
+	params ["_mapIsOpened", ""];
+	if (!_mapIsOpened) exitWith {
+		f_var_mapClickTeleport_MapClosed = true;
+	};
+}];
+
+// ====================================================================================
+
+// TELEPORT FUNCTIONALITY
+// Open the map for the player and display a notification, then set either the player's vehicle
+// or the unit to the new position. If the group needs to be teleported too, set the group's position
+// as well.
+
+["MapClickTeleport",[f_var_mapClickTeleport_textSelect]] call BIS_fnc_showNotification;
+
+["mapClickTeleportEH", "onMapSingleClick", {f_var_mapClickTeleport_telePos = _pos;}] call BIS_fnc_addStackedEventHandler;
+openMap [true, false];
+waitUntil {f_var_mapClickTeleport_MapClosed || !isNil "f_var_mapClickTeleport_telePos"};
+["mapClickTeleportEH", "onMapSingleClick"] call BIS_fnc_removeStackedEventHandler;
+
+removeMissionEventHandler ["Map", f_var_mapClickTeleport_EHMap];
+
+//Exit if the map was closed without selecting a position.
+if (f_var_mapClickTeleport_MapClosed) exitWith {};
+
+//Select units to be teleported
+private _units = [player];
+if (f_var_mapClickTeleport_GroupTeleport) then {
+	_units = units group player;
+};
+
+// Move player/group
+// If the player is in a vehicle and not HALO-ing, the complete vehicle is moved.
+// Otherwise the players are teleported individually.
+
+if (vehicle player != player && f_var_mapClickTeleport_Height == 0) then {
+	(vehicle player) setPos (f_var_mapClickTeleport_telePos findEmptyPosition [0,150,typeOf (vehicle player)]);
```

Comment:
If `f_var_mapClickTeleport_GroupTeleport` = false, any passengers will still be able to teleport themselves (and the vehicle!) after this.

Reply:
yes.

Reply:
I'm not sure we want that...

Reply:
Hmm, I don't like the possibility being there though. Maybe we should move this out of 3.5.2 and have a proper discussion in #f3_fa and #missionmaking about what features are actually used and useful. For example, I can't think so any mission where `f_var_mapClickTeleport_Uses` > 1 or where `f_var_mapClickTeleport_GroupTeleport` = false or even where `f_var_mapClickTeleport_Units` is non-empty.

Eliminating those three parameters would reduce the complexity and resolve this problem entirely (only group leader could teleport)

Reply:
`or where f_var_mapClickTeleport_GroupTeleport = false`
Groups don't always/necessarily fit in a single vehicle though.

Reply:
So it'd do the current behaviour of teleport the extras individually to the same position?